### PR TITLE
[Ingest] Cleanup initialization code

### PR DESCRIPTION
**Pull request by martijnvg**
- Folded IngestModule into NodeModule
- Renamed IngestBootstrapper to IngestService
- Let NodeService construct IngestService and removed the Guice annotations
- Let IngestService implement Closable

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/action/ingest/IngestActionFilterTests.java [hunk 1]
```diff
@@ -63,8 +64,10 @@
     @Before
     public void setup() {
         executionService = mock(PipelineExecutionService.class);
-        IngestBootstrapper bootstrapper = mock(IngestBootstrapper.class);
-        when(bootstrapper.getPipelineExecutionService()).thenReturn(executionService);
+        IngestService ingestService = mock(IngestService.class);
+        when(ingestService.getPipelineExecutionService()).thenReturn(executionService);
+        NodeService bootstrapper = mock(NodeService.class);
```
Comment (javanna) [inline core/src/test/java/org/elasticsearch/action/ingest/IngestActionFilterTests.java]: nit: rename the variable too?

File: core/src/test/java/org/elasticsearch/action/ingest/IngestActionFilterTests.java [hunk 2]
```diff
@@ -170,8 +173,10 @@ public String getType() {
         };
         when(store.get("_id")).thenReturn(new Pipeline("_id", "_description", new CompoundProcessor(processor)));
         executionService = new PipelineExecutionService(store, threadPool);
-        IngestBootstrapper bootstrapper = mock(IngestBootstrapper.class);
-        when(bootstrapper.getPipelineExecutionService()).thenReturn(executionService);
+        IngestService ingestService = mock(IngestService.class);
+        when(ingestService.getPipelineExecutionService()).thenReturn(executionService);
+        NodeService bootstrapper = mock(NodeService.class);
```
Comment (javanna) [inline core/src/test/java/org/elasticsearch/action/ingest/IngestActionFilterTests.java]: nit: rename the variable too?

File: core/src/test/java/org/elasticsearch/ingest/IngestClientIT.java [hunk 3]
```diff
@@ -224,7 +225,7 @@ public String description() {
             return "ingest mock";
         }
 
-        public void onModule(IngestModule ingestModule) {
+        public void onModule(NodeModule ingestModule) {
```
Comment (javanna) [inline core/src/test/java/org/elasticsearch/ingest/IngestClientIT.java]: nit: rename the argument too?

File: core/src/main/java/org/elasticsearch/ingest/IngestService.java [hunk 4]
```diff
@@ -19,50 +19,28 @@
...
+import java.io.Closeable;
 import java.io.IOException;
-import java.io.InputStream;
 
 /**
  * Instantiates and wires all the services that the ingest plugin will be needing.
  * Also the bootstrapper is in charge of starting and stopping the ingest plugin based on the cluster state.
  */
-public class IngestBootstrapper extends AbstractLifecycleComponent {
+public class IngestService implements Closeable {
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/ingest/IngestService.java]: given how small it is, I now wonder if we still need this service... given that we simply do in a couple of places nodeService.getIngestService().get* where * is either the PipelineStore or the PipelineExecutionService. Is it worth to add these two to NodeService directly and get rid of the service class too?

**Comment by javanna** [comment]
left a few comments, looks very good to me already though

File: core/src/main/java/org/elasticsearch/ingest/IngestService.java (hunk 4 above)
Comment (martijnvg) [inline core/src/main/java/org/elasticsearch/ingest/IngestService.java]: This is what I did initially, but then I had to make a few other changes too (like implementing closable on NodeService). It feels good to me to bundle ingest related together, at least for now.

**Comment by martijnvg** [comment]
@javanna I've renamed the variables.

**Comment by javanna** [comment]
LGTM

**Pull request merged**